src/source-map.ts

```diff
@@ -18,7 +19,7 @@ export const installSourceMapSupport = () => {
 		 *
 		 * https://github.com/nodejs/node/blob/91193825551f9301b6ab52d96211b38889149892/lib/internal/errors.js#L141
 		 */
-		&& typeof Error.prepareStackTrace !== 'function'
+		&& (isFeatureSupported(prepareStackTraceExposed) || typeof Error.prepareStackTrace !== 'function')
```

Comment:
Upon reading what [the default `prepareStackTrace`](https://github.com/nodejs/node/blob/dd711d221a6cd3ffaccedf6fc810b9c2d16eedd5/lib/internal/errors.js#L95-L111) looks like, I'm okay with removing this check entirely.

The Node docs previously implied that if it was overridden sourcemaps wouldn't work expectedly, but the implementation is so simple that the default behavior can easily be replicated.

Reply:
are you sure remove checking Error.prepareStackTrace? I don't know the history.

Reply:
Hmm, I dug into it deeper...

Previously, the Node docs said:
> Overriding `Error.prepareStackTrace` prevents `--enable-source-maps` from modifying the stack trace.

And the `prepareStackTrace` logic was:
- no source maps: https://github.com/nodejs/node/blob/v20.11.1/lib/internal/errors.js#L119-L133
- source maps: https://github.com/nodejs/node/blob/v20.11.1/lib/internal/source_map/prepare_stack_trace.js#L32-L110

Not generating source maps when `Error.prepareStackTrace` was set was a performance optimization because I assumed the custom `prepareStackTrace` wouldn't be able to handle the source map. But this may have been an incorrect assumption. Furthermore, it's a rare optimization because I think `Error.prepareStackTrace` is not very common.

That said, it's probably fine to remove this check entirely for how little benefit it provides, and the possibility that a custom `prepareStackTrace` can handle source maps.

Reply:
removed